samples/extensions/shader_debugprintf/shader_debugprintf.cpp

```diff
@@ -406,11 +439,14 @@ std::unique_ptr<vkb::Instance> ShaderDebugPrintf::create_instance(bool headless)
 
 	enabled_extensions.push_back(VK_EXT_DEBUG_UTILS_EXTENSION_NAME);
 	enabled_extensions.push_back(VK_KHR_GET_PHYSICAL_DEVICE_PROPERTIES_2_EXTENSION_NAME);
+#if (defined(VKB_ENABLE_PORTABILITY))
+	enabled_extensions.push_back(VK_KHR_PORTABILITY_ENUMERATION_EXTENSION_NAME);
+#endif
 
 	VkApplicationInfo app_info{VK_STRUCTURE_TYPE_APPLICATION_INFO};
 	app_info.pApplicationName = "Shader debugprintf";
 	app_info.pEngineName      = "Vulkan Samples";
-	app_info.apiVersion       = VK_API_VERSION_1_1;
```

Comment:
The reason behind requiring 1.1 here was a bug in the validation layers. On 1.0 this sample run with less than one frame per second for me during developement, and last time I checked that error was still present and the cause for that validation layer bug hasn't been found. I'd prefer to revert this change because of that.

Reply:
This is quite strange, since I find the opposite problem with macOS when VK_API_VERSION_1_1 is activated - very slow operation.  This slowdown does not occur when VK_API_VERSION_1_0 is activated in the custom override, or as part of the standard framework (which is defaulted to VK_API_VERSION_1_0) when using layer settings instead.

I wil restest this on Windows and Linux using VK_API_VERSION_1_0 to see what happens with both the custom override and with the standard framework using layer settings.  Stay tuned.

Reply:
Sorry, I mixed up. The slowdown ONLY happens with 1.1. I sadly can't remember why I forced 1.1 for this sample, but I think it was another problem with validation layers at that time.

As for your findings (slowdown), can you add that to this issue: https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/7562

Reply:
Thanks for the clarification.  And yes, I will add to the Validation Layers issue you mention above.

Reply:
After testing on linux, I have found that compile time #ifdef logic for selecting between `VK_EXT_layer_settings` and the original `create_instance()` override is not sufficient.  `VK_EXT_layer_settings` can be defined in the included Vulkan Headers but may not be available at runtime depending on the SDK version one has installed.  This can result in failures if the extension is not available.  To get around this I would have to perform a runtime check for `VK_EXT_layer_settings` which adds complexity.  Let me think about this and see if that can be made simple enough.  If not, I may have to revert the layer settings change in the **shader_debugprintf** sample.

Reply:
I added an additional commit that checks for `VK_EXT_layer_settings ` at runtime and falls back to the override code if not available.  Code change was relatively simple, and actually reduces the #ifdefs.  Tested on Windows, Linux, macOS.